### PR TITLE
Redesign custom containers

### LINKED ISSUE #407: Re-theme warning block in dark theme
изображение

### PR DESCRIPTION
**Pull request by CikiMomogi**
Enhance the tip, warning, and danger containers' color schemes. This PR may resolve #407

Before:
grammy-before

After:
gramY-after

### CODE REVIEW

**Comment by roj1512** [comment]
@KnorpelSenf You in favor of it?

**Comment by KnorpelSenf** [comment]
I would like to hear the opinion of @Loskir on this

File: site/docs/.vuepress/styles/index.scss [hunk 1]
```diff
@@ -6,9 +6,52 @@
 }
 
 html.dark {
+  --c-bg-lighter: #2b313a;
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:9]: Hmm, why did you make lighter the same color as light?
image
This makes code snippets inside tables blend in

File: site/docs/.vuepress/styles/index.scss [hunk 2]
```diff
@@ -6,9 +6,52 @@
...
   --c-brand-light: #138cb3;
-  --c-tip: #1583a7;
   --c-brand: #009dca;
+  --c-tip: #1583a7;
+  --c-tip-title: var(--c-text-accent);
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:13]: This makes tip title appear in bright blue color (as opposed to text color by default)

Tips are not consistent with danger and warning blocks, but we can fix it. Anyway, that should be moved to a separate PR, as it requires changes to light theme too

File: site/docs/.vuepress/styles/index.scss [hunk 3]
```diff
@@ -6,9 +6,52 @@
...
+  --c-warning: #e0ad15;
+  --c-warning-bg: rgba(255, 197, 23, .05);
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:15]: Would be nice to make these colors solid instead of transparent

File: site/docs/.vuepress/styles/index.scss [hunk 4]
```diff
@@ -6,9 +6,52 @@
...
+
+html.dark .custom-container.warning code{
+  color: rgba(252, 210, 83, 0.9);
+  background-color: rgba(255, 197, 23, .1);
+}
+
+html.dark .custom-container.danger code{
+  color: rgb(252, 83, 83);
+  background-color: rgba(237, 60, 80, .15);
+}
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:32]: These changes to code blocks definitely add a nice touch, but you forgot the light theme too. I suggest moving it to a separate PR

File: site/docs/.vuepress/styles/index.scss [hunk 5]
```diff
@@ -6,9 +6,52 @@
...
+
+html.dark .custom-container.warning a{
+  color: var(--c-warning);
+}
+
+html.dark .custom-container.danger a{
+  color: rgb(253, 54, 54);
+}
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:40]: Same here. Also not very consistent with the way default theme applies styles to links. It should be `--c-danger-text-accent` and `--c-warning-text-accent`

File: site/docs/.vuepress/styles/index.scss [hunk 6]
```diff
@@ -6,9 +6,52 @@
...
+
+html.dark .custom-container.warning blockquote{
+  color: rgba(252, 252, 124, 0.65);
+}
+
+html.dark .custom-container.danger blockquote{
+  color: rgb(253, 128, 136, 0.7);
+}
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:48]: Same here

File: site/docs/.vuepress/styles/index.scss [hunk 7]
```diff
@@ -6,9 +6,52 @@
...
+
+.custom-container.tip pre code, .custom-container.tip pre[class*="language-"] code,
+.custom-container.warning pre code, .custom-container.warning pre[class*="language-"] code,
+.custom-container.danger pre code, .custom-container.danger pre[class*="language-"] code {
+  color: #fff;
+  background-color: transparent;
```
Comment (Loskir) [inline site/docs/.vuepress/styles/index.scss:54]: These changes seem to copy the default styles. As far as I can tell, they are not needed, if you write other styles in a correct way

**Review by Loskir — CHANGES_REQUESTED** [review]
Thank you for your contribution! It looks amazing!

However, there are some comments

**Comment by Loskir** [comment]
I've played around with these styles a bit and I want to contribute my changes. I think I won't be able to commit here directly, so I'll create a PR to @CikiMomogi's repo

File: site/docs/.vuepress/styles/index.scss (hunk 1 above)
Comment (CikiMomogi) [reply site/docs/.vuepress/styles/index.scss:9]: Oops... My bad.

File: site/docs/.vuepress/styles/index.scss (hunk 2 above)
Comment (CikiMomogi) [reply site/docs/.vuepress/styles/index.scss:13]: I believe it will be more appealing if the title text and accent color are the same color. I even considered modifying the container's background color to match the accent color, as with the warning and danger containers. What is your opinion?

**Comment by CikiMomogi** [comment]
https://grammy-website.web.app/test.html

**Comment by KnorpelSenf** [comment]
Looks pretty cool!

**Comment by KnorpelSenf** [comment]
Looking forward to the pending suggestions, and to see this merged.

We may want to ping the people at VuePress and suggest to upstream your contributions.

**Comment by CikiMomogi** [comment]
Hey, it's been two months since then. Any updates for this PQ?

**Comment by roj1512** [comment]
@CikiMomogi You know. There are unresolved reviews.

**Comment by CikiMomogi** [comment]
> @CikiMomogi You know. There are unresolved reviews.

I know. The suggestion is already implemented, but I'm waiting for @Loskir reply to reconsider it.

**Comment by roj1512** [comment]
@Loskir
